lib/errors.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+class TldrError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'TldrError';
+  }
+
+  static isTldrError(err) {
+    return err instanceof this;
+  }
+}
+
+class EmptyCacheError extends TldrError {
+  constructor() {
+    super(trim`
+Local cache is empty
+Please run tldr --update
+    `);
+    this.name = 'EmptyCacheError';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 2;
+  }
+}
+
+class MissingPageError extends TldrError {
+  constructor(repo) {
+    super(trim`
+Page not found.
+If you want to contribute it, feel free to send a pull request to: ${repo}
+    `);
+    this.name = 'MissingPageErrror';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 3;
+  }
+}
+
+module.exports = {
+  TldrError,
+  EmptyCacheError,
+  MissingPageError
+};
+
+function trim(strings, ...values) {
+  let output = values.reduce((acc, value, i) => {
+    return acc + strings[i] + value;
+  }, '') + strings[values.length];
```

Comment:
@vladimyr is there a reason you didn't do this?
```suggestion
  let output = values.reduce((acc, value, i) => acc + strings[i] + value, '') + strings[values.length];
```

Reply:
Unfortunately, ESLint wouldn't let me do it 🙁
https://github.com/tldr-pages/tldr-node-client/blob/7e49dda42a11dfc07b5547b1bc9a20bc334b8792/.eslintrc.json#L9

Reply:
@MasterOdin / @agnivade do you have any preferences regarding this style?  If you don't I'd like to relax this rule.

Reply:
My guess is it will effectively double down the LOC amount...

Reply:
Let's wait for a different time.